### PR TITLE
#162949233 Article highlighting with comments

### PR DESCRIPTION
**Pull request by sylviawanjiku**
#### What does this PR do?
Implements highlighting of articles and allows commenting on the highlighted section
#### Description of Task to be completed?

-Highlight article
-Comment on the highlighted article
-crud for the comment on the highlighted part of the article

#### How should this be manually tested?

1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Cd into the cloned repo 
3. Git checkout to branch `ft-highlight-comment-article-162949233`
4. Set up a virtual environment and install requirements:
      ```
            python3 -m venv venv
            source venv/bin/activate
            pip3 install -r requirements.txt

     ```
5. Set up a .env file on the same level as the venv file and add the following:
      ```
            source venv/bin/activate
            export SECRET_KEY='yoursecretkey'
            export DATABASE_URL='postgres://yourdatabasename'
            export HOST_PASSWORD=" "
            export HOST_USER=" "
            export SMTP_HOST=' '
            export EMAIL_PORT=''

      ```
6. Then run: 
     ``` 
            source .env 
            ./manage.py migrate
            ./manage.py runserver
      ```

7. Fire up postman and test the following endpoints:

| EndPoint                       | Functionality                           |
| ---------------------- |:----------------------------:|
| POST     api/articles/ `<slug>`/highlight  | Highlight and comment on articlel                      |
| GET     api/articles/ `<slug>`/highlight  | View all highlight comments specific to an articlel                      |
| GET     api/articles/`<slug>`/highlight/`<int:id> `                  |Get a highlight comment  |
| PUT     api/articles/`<slug>`/highlight/`<int:id> `                  |Update a highlight comment  |
| DELETE     api/articles/`<slug>`/highlight/`<int:id> `                |Delete a highlight comment  |

                                                    
use the following payload format for sharing an article via email:
```
{
    "comment": " {
                "body": " ",
                "begin_index": " ",
                "end_index": " "
            }"
  }
```

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[162949233]https://www.pivotaltracker.com/story/show/162949233
#### Screenshots (if appropriate)
image

#### Questions:

### CODE REVIEW

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -183,7 +179,32 @@ class CommentHistorySerializer(serializers.ModelSerializer):
     """
     This class handles the history of the comment edited
     """
-
     class Meta:
         model = Comments
         fields = ('id', 'body', 'created_at', 'updated_at')
+
```
Comment (MbugwaSami) [inline authors/apps/articles/serializers.py]: please add another line before  line 186 to adhere to PEP8 convention of two lines before a class.

File: authors/apps/articles/views/articles.py [hunk 2]
```diff
@@ -193,3 +199,114 @@ class TagView(ListAPIView):
     renderer_classes = (TagJSONRenderer,)
     serializer_class = CustomTagSerializer
     queryset = Tag.objects.all()
+
```
Comment (MbugwaSami) [inline authors/apps/articles/views/articles.py]: same here, remove one empty line.

**Review by MbugwaSami — DISMISSED** [review]

**Review by MbugwaSami — DISMISSED** [review]

File: authors/apps/articles/views/articles.py [hunk 3]
```diff
@@ -331,3 +338,113 @@ def post(self, request, *args, **kwargs):
...
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class RetrieveUpdateDeleteHighlight(RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly, )
+    serializer_class = HighlightArticleSerializer
+    queryset = HighlightArticleModel.objects.all()
+    lookup_field = 'id'
+
+    def get(self, request, slug, id):
```
Comment (Allan690) [inline authors/apps/articles/views/articles.py:405]: id is a reserved keyword in python. You can use article_id instead.

**Review by Allan690 — COMMENTED** [review]
Kindly make the changes requested

**Review by nesh-dev — DISMISSED** [review]
Good work handles the slug issue as well.

File: authors/tests/test_articles.py [hunk 4]
```diff
@@ -149,40 +149,27 @@ def test_delete_article(self):
...
         response = self.client.delete(
             reverse(
                 'articles:detail_article',
-                kwargs={'slug': 'another-posts'},
+                kwargs={'slug': "hshshs"},
             ),
             content_type='application/json'
         )
         self.assertIn(error_messages['article_404'], response.data)
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```
Comment (Allan690) [inline authors/tests/test_articles.py:163]: Why was this test removed?

File: authors/tests/test_articles.py [hunk 5]
```diff
@@ -149,40 +149,27 @@ def test_delete_article(self):
...
 
-    def test_delete_unauthorized(self):
```
Comment (Allan690) [inline authors/tests/test_articles.py:164]: Why was this test removed?

**Review by kwanj-k — DISMISSED** [review]

File: authors/apps/articles/views/articles.py (hunk 2 above)
Comment (sylviawanjiku) [reply authors/apps/articles/views/articles.py]: Done

**Review by kwanj-k — APPROVED** [review]

**Review by nesh-dev — APPROVED** [review]

**Pull request merged**